pkg/cmd/server/origin/master.go

```diff
@@ -492,6 +496,27 @@ func (c *MasterConfig) GetRestStorage() map[string]rest.Storage {
 		storage["builds/log"] = buildlogregistry.NewREST(buildStorage, buildStorage, c.BuildLogClient(), kubeletClient)
 	}
 
+	if sdnfactory.IsMultitenantNetworkPlugin(c.Options.NetworkConfig.NetworkPluginName) {
+		netNamespaceStorage := netnamespaceetcd.NewREST(c.EtcdHelper)
+		netNamespaceRegistry := netnamespace.NewRegistry(netNamespaceStorage)
+		c.MultitenantNetworkConfig.NetNamespaceRegistry = netNamespaceRegistry
+
+		netIDRange, err := sdnutil.NewVNIDRange(sdnutil.MinVNID, sdnutil.MaxVNID-sdnutil.MinVNID+1)
+		if err != nil {
+			glog.Fatalf("Unable to create NetID range: %v", err)
+		}
+		c.MultitenantNetworkConfig.NetIDRange = *netIDRange
+
+		netIDAllocator := vnidallocator.NewAllocatorCustom(*netIDRange, func(max int, rangeSpec string) allocator.Interface {
```

Comment:
Does the controller depend on having the exact same instance of this in order for the API and the repair controller to interact correctly? If so, won't that cause problems with HA master, where you have several API handlers with their own allocators, and one controller with an allocator?